kubediff

```diff
@@ -1,50 +1,97 @@
 #!/usr/bin/env python
+# -*- coding: utf-8 -*-
 
-import optparse
+from __future__ import (division, absolute_import, print_function,
+                        unicode_literals)
+
+import argparse
 import sys
 
 from kubedifflib import (
-  check_files,
-  JSONPrinter,
-  QuietTextPrinter,
+    check_files,
+    JSONPrinter,
+    QuietTextPrinter,
 )
 
 
+class ParseArgs():
+    def __init__(self):
+
+        # Parse arguments passed at cli
+        self.parse_arguments()
+
+    def parse_arguments(self):
+
+        description = '''
+     _          _             _  _   __   __
+    | |__ _  _ | |__  ___  __| |(_) / _| / _|
+    | / /| || || '_ \/ -_)/ _` || ||  _||  _|
+    |_\_\ \_,_||_.__/\___|\__,_||_||_|  |_|
+
+    Compare yaml files in path to running state in kubernetes and print the
+    differences. This is useful to ensure you have applied all your changes
+    to the appropriate environment. This tools runs kubectl, so unless your
+    ~/.kube/config is configured for the correct environment, you will need
+    to supply the kubeconfig for the appropriate environment.'''
+
+        parser = argparse.ArgumentParser(description=description,
+                                         formatter_class=argparse.RawTextHelpFormatter)
+
+        parser.add_argument('--kubeconfig',
+                            '-k',
+                            help='path to kubeconfig')
+
+        parser.add_argument('--context',
+                            '-c',
+                            help='name of kubeconfig context to use')
+
+        parser.add_argument('--namespace',
+                            '-n',
+                            help=('Namespace to assume for objects where it is not '
+                                  'specified (default = Kubernetes default for '
+                                  'current context)'))
```

Comment:
Does a default value need to be specified here?

Reply:
I have kept as the original, but I think is a great idea. Let me implement it.

Reply:
Added default namespace as default :D